### PR TITLE
add supporting data properties fixes #853 fixes #845 fixes #1074

### LINKED ISSUE #845: Create edge properties for 'supporting data' types
The Biolink/SRI team will be create edge properties for to holds the different types of 'supporting data' that hang from Edges in Translator KGs. 
 
As per discussions on recent EPC and Data Modeling calls, we need KPs and ARAs to [go to the spreadsheet here](https://docs.google.com/spreadsheets/d/1ckc37my-L_y7rhX0S4j1aotz1dWRs_bJhhDiUGYnZW4/edit#gid=0) and enter information described in columns F, G, H and I for each type of supporting data they need to represent. This info will allow us to create an edge property for you to use in your data in Attributes holding these types of supporting data. 
 
Background about 'supporting data' and our approach and requirements for modeling are documented in the [google document here](https://docs.google.com/document/d/1YHsSSn8ML0iPJg9-3XsqMq-1eiFiPih8qij7h-26BKo/edit#). 
 
The columns are self-describing, and should provide clear instructions (along with the background documentation linked above). But please reach out to @mbrush in this ticket, or Slack if you have questions or feedback.

### LINKED ISSUE #853: Automate the addition of spreadsheet metamodels (supporting data sources in this ticket) into biolink-model
Building off #606, sometimes we have large sets of node and edge properties to add to the model.  See the [supporting data spreadsheet of properties](https://docs.google.com/spreadsheets/d/1ckc37my-L_y7rhX0S4j1aotz1dWRs_bJhhDiUGYnZW4/edit#gid=0) that need to be added.

We have a couple of different code bases to take inspiration (and possible reuse code) from:
https://github.com/microbiomedata/metadata_converter  <-- converts large spreadsheets of properties into linkml YAML. 
https://github.com/linkml/linkml-runtime/blob/main/linkml_runtime/loaders/csv_loader.py

Automating the "supporting data" spreadsheet gives us a chance to reflect on the format of these files and structure them so that large model additions can happen quickly and accurately according to requirements.

### LINKED ISSUE #1074: supporting study results for COHD
**Is your feature request related to a problem? Please describe.**
Matt B. helped us to develop a 2-level `supporting study result` structure to model COHD data, but most of the proposed elements are not defined in Biolink yet. 
[Google doc](https://docs.google.com/document/d/1YHsSSn8ML0iPJg9-3XsqMq-1eiFiPih8qij7h-26BKo/edit#heading=h.j4lrz7khxrbl) describing the model. Matt's [examples ](https://github.com/NCATSTranslator/Evidence-Provenance-Confidence-Working-Group/blob/master/attribute_epc_examples/COHD_TRAPI1.2_nested_attributes.yml)(see the example set at the bottom)

**What working group (or team) did this request originate from?**
Clinical Data Services (COHD)

**Note:** This is relevant for members of NCATS Translator.

**Describe the solution you'd like**
The following may need to be defined in Biolink:
* biolink:supporting_study_result
* biolink:ConceptCountAnalysisResult
* biolink:concept_pair_count
* biolink:concept_count_subject
* biolink:concept_count_object
* biolink:supporting_dataset
* biolink:ChiSquaredAnalysisResult
* biolink:unadjusted_p-value
* biolink:bonferonni_adjusted_p-value
* biolink:Observed-ExpectedFrequencyAnalysisResult
* biolink:expected_count
* biolink:ln_ratio
* biolink:ln_ratio_99_confidence_interval
* biolink:RelativeFrequencyAnalysisResult
* biolink:relative_frequency_subject
* biolink:relative_freq_subject_confidence_interval
* biolink:relative_frequency_object
* biolink:relative_freq_object_confidence_interval

**Additional information to support this request** (optional)
We've recently implemented this model in the COHD's dev endpoint at https://cohd.io/api/query and we'd like to push these changes out to the ITRB environments soon (ideally before the Aug 29 code freeze). From our perspective, we're not in a rush to have these defined immediately in Biolink as it appears that reasoners are still able to utilize the data (e.g., ARAX UI is able to display the results). We are open to guidance on how to move forward with this. 

Sample query that can be sent to COHD to see results:
```
{
    "message": {
        "query_graph": {
            "nodes": {
                "subj": {
                    "ids": ["MONDO:0009061"]
                },
                "obj": {
                    "categories": ["biolink:DiseaseOrPhenotypicFeature"]
                }
            },
            "edges": {
                "e0": {
                    "subject": "subj",
                    "object": "obj",
                    "predicates": ["biolink:has_real_world_evidence_of_association_with"]
                }
            }
        }
    }
}
```

Sample attributes from a single edge:
```
                    "attributes": [
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:original_knowledge_source",
                            "description": "The COHD KP defines associations between biomedical concepts based on statistical analysis of clinical/EHR data.",
                            "value": "infores:cohd",
                            "value_type_id": "biolink:InformationResource",
                            "value_url": "http://cohd.io/api/query"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_dataset",
                            "description": "Dataset ID within COHD",
                            "original_attribute_name": "dataset_id",
                            "value": "COHD:dataset_1",
                            "value_type_id": "EDAM:data_1048"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_study_result",
                            "attributes": [
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:concept_pair_count",
                                    "description": "Observed concept count between the pair of subject and object nodes",
                                    "original_attribute_name": "concept_pair_count",
                                    "value": 449,
                                    "value_type_id": "EDAM:data_0006"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:concept_count_subject",
                                    "description": "Observed concept count of the subject node",
                                    "original_attribute_name": "concept_count_subject",
                                    "value": 968,
                                    "value_type_id": "EDAM:data_0006"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:concept_count_object",
                                    "description": "Observed concept count of the object node",
                                    "original_attribute_name": "concept_count_object",
                                    "value": 556,
                                    "value_type_id": "EDAM:data_0006"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:supporting_dataset",
                                    "description": "Dataset ID within COHD",
                                    "original_attribute_name": "dataset_id",
                                    "value": "COHD:dataset_1",
                                    "value_type_id": "EDAM:data_1048"
                                }
                            ],
                            "description": "A study result describing the initial count of concepts",
                            "value": null,
                            "value_type_id": "biolink:ConceptCountAnalysisResult"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_study_result",
                            "attributes": [
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:unadjusted_p-value",
                                    "description": "Chi-square p-value, unadjusted. http://cohd.io/about.html",
                                    "original_attribute_name": "p-value",
                                    "value": 1e-12,
                                    "value_type_id": "EDAM:data_1669",
                                    "value_url": "http://edamontology.org/data_1669"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:bonferonni_adjusted_p-value",
                                    "description": "Chi-square p-value, Bonferonni adjusted by number of pairs of concepts. http://cohd.io/about.html",
                                    "original_attribute_name": "p-value adjusted",
                                    "value": 1e-12,
                                    "value_type_id": "EDAM:data_1669",
                                    "value_url": "http://edamontology.org/data_1669"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:supporting_dataset",
                                    "description": "Dataset ID within COHD",
                                    "original_attribute_name": "dataset_id",
                                    "value": "COHD:dataset_1",
                                    "value_type_id": "EDAM:data_1048"
                                }
                            ],
                            "description": "A study result describing a chi-squared analysis on a single pair of concepts",
                            "value": null,
                            "value_type_id": "biolink:ChiSquaredAnalysisResult"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_study_result",
                            "attributes": [
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:expected_count",
                                    "description": "Calculated expected count of concept pair. For ln_ratio. http://cohd.io/about.html",
                                    "original_attribute_name": "expected_count",
                                    "value": 0.3006024806317585,
                                    "value_type_id": "EDAM:operation_3438"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:ln_ratio",
                                    "description": "Observed-expected frequency ratio. http://cohd.io/about.html",
                                    "original_attribute_name": "ln_ratio",
                                    "value": 7.308989437171575,
                                    "value_type_id": "EDAM:data_1772"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:ln_ratio_99_confidence_interval",
                                    "description": "Observed-expected frequency ratio 0.99% confidence interval",
                                    "original_attribute_name": "ln_ratio_confidence_interval",
                                    "value": [
                                        7.1447659245560216,
                                        7.455795361004793
                                    ],
                                    "value_type_id": "EDAM:data_0951"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:supporting_dataset",
                                    "description": "Dataset ID within COHD",
                                    "original_attribute_name": "dataset_id",
                                    "value": "COHD:dataset_1",
                                    "value_type_id": "EDAM:data_1048"
                                }
                            ],
                            "description": "A study result describing an observed-expected frequency anaylsis on a single pair of concepts",
                            "value": null,
                            "value_type_id": "biolink:Observed-ExpectedFrequencyAnalysisResult"
                        },
                        {
                            "attribute_source": "infores:cohd",
                            "attribute_type_id": "biolink:supporting_study_result",
                            "attributes": [
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:relative_frequency_subject",
                                    "description": "Relative frequency, relative to the subject node. http://cohd.io/about.html",
                                    "original_attribute_name": "relative_frequency_subject",
                                    "value": 0.46384297520661155,
                                    "value_type_id": "EDAM:data_1772"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:relative_freq_subject_confidence_interval",
                                    "description": "Relative frequency (subject) 0.99% confidence interval",
                                    "original_attribute_name": "relative_freq_subject_confidence_interval",
                                    "value": [
                                        0.3765490943755958,
                                        0.5680539932508436
                                    ],
                                    "value_type_id": "EDAM:data_0951"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:relative_frequency_object",
                                    "description": "Relative frequency, relative to the object node. http://cohd.io/about.html",
                                    "original_attribute_name": "relative_frequency_object",
                                    "value": 0.8075539568345323,
                                    "value_type_id": "EDAM:data_1772"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:relative_freq_object_confidence_interval",
                                    "description": "Relative frequency (object) 0.99% confidence interval",
                                    "original_attribute_name": "relative_freq_object_confidence_interval",
                                    "value": [
                                        0.63915857605178,
                                        1.0181451612903225
                                    ],
                                    "value_type_id": "EDAM:data_0951"
                                },
                                {
                                    "attribute_source": "infores:cohd",
                                    "attribute_type_id": "biolink:supporting_dataset",
                                    "description": "Dataset ID within COHD",
                                    "original_attribute_name": "dataset_id",
                                    "value": "COHD:dataset_1",
                                    "value_type_id": "EDAM:data_1048"
                                }
                            ],
                            "description": "A study result describing a relative frequency anaylsis on a single pair of concepts",
                            "value": null,
                            "value_type_id": "biolink:RelativeFrequencyAnalysisResult"
                        }
                    ]
```

Also, Text Mining Provider uses a similar structure and may need similar support. See the Google doc linked above, and [Bill's slides from the KP of the Month](https://docs.google.com/presentation/d/1e5ftEHO4x_PkW4Q_mHBmwVqEzaXKLyMYJFs52nw4UoE/edit#slide=id.g147d2142526_0_302) presentation. 

**Tag relevant members for discussion**
@mbrush @bill-baumgartner

### PR DESCRIPTION
**Pull request by sierra-moxon**

### CODE REVIEW

File: biolink-model.yaml [hunk 1]
```diff
@@ -6145,6 +6145,717 @@ slots:
...
+      same document (or part of a document) in a text corpus.
+    range: string
+
+  contribution towards patient survival:
+    is_a: association slot
+    description: >-
+      Contribution is a summary statistic indicating the level of contribution
+      an entity has on patient survival time as measured by its quantifiable impact
+      on a distribution of patient survival time.
+    range: integer
```
Comment (cmungall) [inline biolink-model.yaml:6404]: I think this is a typo and you mean float

But the first example is a range and not a float...

Comment (cmungall) [reply biolink-model.yaml:6404]: We may also want to define our own types here. Can also indicate bounds, e.g. between 0 and 1

File: biolink-model.yaml [hunk 2]
```diff
@@ -6145,6 +6145,717 @@ slots:
...
+  half maximal inhibitory concentration:
+    is_a: association slot
+    description: IC50 indicates how much drug is needed to inhibit a biological process
+      by half, thus providing a measure of potency of an antagonist drug in pharmacological
+      research
+    range: datetime
+    examples:
+      - value: '-2.112848714'
+
+  binding constant:
```
Comment (cmungall) [inline biolink-model.yaml:6426]: We can add mappings here https://bioportal.bioontology.org/search?utf8=%E2%9C%93&query=binding+constant&button=

File: biolink-model.yaml [hunk 3]
```diff
@@ -6145,6 +6145,717 @@ slots:
...
+    description: The Kd is a useful way to present the affinity of a drug for its
+      biological target. Kd tells us the concentration of drug that is required to
+      yield a significant amount of interaction with the target protein.
+    range: integer
+    examples:
+      - value: '6800'
+
+  z score:
+    is_a: association slot
+    description: zscore is the measure of how many standard deviations below or above
```
Comment (cmungall) [inline biolink-model.yaml:6452]: add space between z and score?

File: biolink-model.yaml [hunk 4]
```diff
@@ -6145,6 +6145,717 @@ slots:
...
+      the population mean, a raw score is.
+    range: datetime
```
Comment (cmungall) [inline biolink-model.yaml:6454]: oops!

File: biolink-model.yaml [hunk 5]
```diff
@@ -6145,6 +6145,717 @@ slots:
...
+  dissociation constant:
+    is_a: association slot
+    description: The Kd is a useful way to present the affinity of a drug for its
+      biological target. Kd tells us the concentration of drug that is required to
+      yield a significant amount of interaction with the target protein.
+    range: integer
+    examples:
+      - value: '6800'
+
+  z score:
```
Comment (cmungall) [inline biolink-model.yaml:6450]: some mappings here https://bioportal.bioontology.org/search?q=Z%20score&ontologies=&include_properties=false&include_views=false&includeObsolete=false&require_definition=false&exact_match=false&categories=

you can use mark's tool to auto-infer these

I would take STATO as primary. They also have exemplar R and scipy functions which help ground these

File: biolink-model.yaml (hunk 3 above)
Comment (sierra-moxon) [reply biolink-model.yaml:6452]: I think TRAPI doesn't support hyphens, `p value` is also a space. 🤔

**Comment by sierra-moxon** [comment]
I think I fixed all the minor changes - will check out Mark's tool next for auto-inference of mappings.

**Comment by sierra-moxon** [comment]
Per discussion with EPC group, closing this for now with the notion that a subsequent effort to reconcile properties will happen.  I will pull out the changes for #1074 into a new PR.

**Pull request closed without merging**